tests/Unit/Api/Command/PayByVisaMobileHandlerTest.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\CommerceWeavers\SyliusTpayPlugin\Unit\Api\Command;
+
+use CommerceWeavers\SyliusTpayPlugin\Api\Command\PayByVisaMobile;
+use CommerceWeavers\SyliusTpayPlugin\Api\Command\PayByVisaMobileHandler;
+use CommerceWeavers\SyliusTpayPlugin\Payum\Factory\CreateTransactionFactoryInterface;
+use CommerceWeavers\SyliusTpayPlugin\Payum\Request\Api\CreateTransaction;
+use Payum\Core\GatewayInterface;
+use Payum\Core\Model\GatewayConfigInterface;
+use Payum\Core\Payum;
+use PHPUnit\Framework\TestCase;
+use Prophecy\Argument;
+use Prophecy\PhpUnit\ProphecyTrait;
+use Prophecy\Prophecy\ObjectProphecy;
+use Sylius\Component\Core\Model\PaymentInterface;
+use Sylius\Component\Core\Model\PaymentMethodInterface;
+use Sylius\Component\Core\Repository\PaymentRepositoryInterface;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Webmozart\Assert\InvalidArgumentException;
+
+class PayByVisaMobileHandlerTest extends TestCase
+{
+    use ProphecyTrait;
+
+    private PaymentRepositoryInterface|ObjectProphecy $paymentRepository;
+
+    private Payum|ObjectProphecy $payum;
+
+    private CreateTransactionFactoryInterface|ObjectProphecy $createTransactionFactory;
+
+    protected function setUp(): void
+    {
+        $this->paymentRepository = $this->prophesize(PaymentRepositoryInterface::class);
+        $this->payum = $this->prophesize(Payum::class);
+        $this->createTransactionFactory = $this->prophesize(CreateTransactionFactoryInterface::class);
+    }
+
+    public function test_it_throw_an_exception_if_a_payment_cannot_be_found(): void
+    {
+        $this->expectException(NotFoundHttpException::class);
+        $this->expectExceptionMessage('Payment with id "1" cannot be found.');
+
+        $this->paymentRepository->find(1)->willReturn(null);
+
+        $this->createTestSubject()->__invoke(new PayByVisaMobile(1, visaMobilePhoneNumber: '44123456789'));
+    }
+
+    public function test_it_throws_an_exception_if_a_gateway_name_cannot_be_determined(): void
+    {
+        $this->expectException(\InvalidArgumentException::class);
+        $this->expectExceptionMessage('Gateway name cannot be determined.');
+
+        $payment = $this->prophesize(PaymentInterface::class);
+        $payment->getDetails()->willReturn([]);
+        $payment->setDetails(Argument::any());
```

Comment:
is it even required?

Reply:
🤷 

![image](https://github.com/user-attachments/assets/a0afe5b7-d997-40bd-b8df-4dbd93d536b2)